lib/signaling/v2/peerconnection.js

```diff
@@ -611,8 +612,12 @@ class PeerConnectionV2 extends StateMachine {
         offer = workaroundIssue8329(offer);
       }
 
+      const sdp = isUnifiedPlan && this._peerConnection.remoteDescription
+        ? unifiedPlanfilterLocalCodecs(offer.sdp, this._peerConnection.remoteDescription.sdp)
+        : offer.sdp;
+
```

Comment:
consider logging orignalSdp.length, and updatedSdp.length to insights publisher.

Reply:
@makarandp0 In order to do this, we have to update the Insights Publisher payload spec and the back-end has to update their data checks so this payload doesn't get rejected.

Reply:
Should we limit the application of this filtering to Group Rooms only (ice-lite agents)? In a Peer-to-Peer Room, another Participant could change their codec support during the life of the PeerConnection (including in an Answer). A JS Client with these changes might artificially restrict the codecs available in their offer, resulting in a preferred codec not being selected (without an additional re-offer). I don't want to write re-offer logic if possible.